sql/core/src/test/scala/org/apache/spark/sql/internal/SQLConfSuite.scala

```diff
@@ -280,4 +280,34 @@ class SQLConfSuite extends QueryTest with SharedSQLContext {
 
     spark.sessionState.conf.clear()
   }
+
+  test("SPARK-22779: correctly compute default value for fallback configs") {
+    val fallback = SQLConf.buildConf("spark.sql.__test__.spark_22779")
+      .fallbackConf(SQLConf.PARQUET_COMPRESSION)
```

Comment:
In SQLConf, all our conf are `TypedConfigBuilder`, instead of `ConfigBuilder`. The type-safe ConfigBuilder is unable to call `fallbackConf `, right?

Reply:
Yea sounds like this PR is fixing a non-existing issue.

Reply:
It's true no existing config uses this, but Reynold was trying to fix this so my guess is he's trying to add one and it wasn't working as expected.

Reply:
I think at least this is good for renaming. In the future, if we rename the SQLConf, we can use this mechanism.